kernel/memory_structs/src/lib.rs

```diff
@@ -24,6 +26,53 @@ use paste::paste;
 use derive_more::*;
 use range_inclusive::{RangeInclusive, RangeInclusiveIterator};
 
+/// Numeric constants associated with different page types.
+pub const PAGE_4KB_SIZE: usize = 1 << 12;
+pub const PAGE_2MB_SIZE: usize = (1 << 12) * 512;
+pub const PAGE_1GB_SIZE: usize = (1 << 12) * 512 * 512;
+
+/// Enum used to indicate the size of a page or frame.
+#[derive(Debug)]
+pub enum MemChunkSize {
+    Normal4K,
+    Huge2M,
+    Huge1G,
+}
+
+/// Trait used to group the size marker structs that paging-related types are parameterized with.
+pub trait PageSize: Ord + PartialOrd + Clone {
+    const SIZE: MemChunkSize;
+    const CONTAINED_4K: usize;
+    const SIZE_IN_BYTES: usize;
+}
+
+/// Marker struct used to indicate the default page size of 4KiB.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub struct Page4KiB;
+impl PageSize for Page4KiB {
+    const SIZE: MemChunkSize = MemChunkSize::Normal4K;
+    const CONTAINED_4K: usize = 1;
+    const SIZE_IN_BYTES: usize = PAGE_SIZE;
+}
+
+/// Marker struct used to indicate a page size of 2MiB.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub struct Page2MiB;
+impl PageSize for Page2MiB {
+    const SIZE: MemChunkSize = MemChunkSize::Huge2M;
+    const CONTAINED_4K: usize = 512;
+    const SIZE_IN_BYTES: usize = PAGE_SIZE * 512;
```

Comment:
these consts (and the ones below in `Page1GiB`) are redundant with the const defintiions up top. You can either re-use the ones defined up top here, or just use these consts as the only consts and remove the ones up top.